x/crosschain/keeper/verify_block_header.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func (k Keeper) VerifyProof(ctx sdk.Context, proof *common.Proof, hash string, txIndex int64, chainID int64) (ethtypes.Transaction, error) {
+	var txx ethtypes.Transaction
+	crosschainFlags, found := k.zetaObserverKeeper.GetCrosschainFlags(ctx)
+	if !found {
+		return txx, fmt.Errorf("crosschain flags not found")
+	}
+	if common.IsBitcoinChain(chainID) && !crosschainFlags.BlockHeaderVerificationFlags.IsBtcTypeChainEnabled {
```

Comment:
We should have a check if `crosschainFlags.BlockHeaderVerificationFlags != nil`
(and I think consider it disabled if `nil`)

Reply:
sounds good , added